sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/buildfeatures/BranchesAndPullRequestsParametersPreprocessor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2000-2020 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jetbrains.buildserver.sonarplugin.buildfeatures;
+
+import java.util.Map;
+
+import org.springframework.util.StringUtils;
+
+import com.google.gson.JsonObject;
+
+import jetbrains.buildServer.messages.Status;
+import jetbrains.buildServer.serverSide.ParametersPreprocessor;
+import jetbrains.buildServer.serverSide.SRunningBuild;
+import jetbrains.buildServer.serverSide.buildLog.MessageAttrs;
+
+/**
+ * Provides SONARQUBE_SCANNER_PARAMS environment variable on build if feature activated
+ */
+public class BranchesAndPullRequestsParametersPreprocessor implements ParametersPreprocessor {
+
+    static final String SQS_SYSENV = "env.SONARQUBE_SCANNER_PARAMS";
+
+    @Override
+    public void fixRunBuildParameters(SRunningBuild build, Map<String, String> runParameters, Map<String, String> buildParams) {
+
+        if (build.getBuildFeaturesOfType(BranchesAndPullRequestsBuildFeature.BUILD_FEATURE_TYPE).isEmpty()) {
+            // Currently only GitHub is supported => if defined, it is for GitHub
```

Comment:
No answer to [15843](https://community.sonarsource.com/t/teamcity-sonarqube-pull-request-analysis-integration-sonarqube-scanner-params-details/15843) 😢 => more simple for a first implementation